applications/SeismicApplication/custom_conditions/fiber_beam_column_section.cpp

```diff
@@ -0,0 +1,192 @@
+//  KRATOS  ____       _               _
+//         / ___|  ___(_)___ _ __ ___ (_) ___
+//         \___ \ / _ \ / __| '_ ` _ \| |/ __|
+//          ___) |  __/ \__ \ | | | | | | (__
+//         |____/ \___|_|___/_| |_| |_|_|\___|
+//
+//  License:     BSD License
+//  license:     structural_mechanics_application/license.txt
+//
+//  Main authors: Mahmoud Zidan
+//    Co authors: Long Chen
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "seismic_application_variables.h"
+#include "custom_conditions/fiber_beam_column_section.hpp"
+
+namespace Kratos {
+
+FiberBeamColumnSection::FiberBeamColumnSection(IndexType NewId)
+    : mId(NewId) {}
+
+FiberBeamColumnSection::FiberBeamColumnSection(
+    IndexType NewId,
+    IntegrationPointType integrationPoint,
+    PropertiesType::Pointer pProperties)
+        : mId(NewId),
+          mPosition(integrationPoint(0)),
+          mWeight(integrationPoint.Weight())
+{
+    KRATOS_TRY
+    mTolerance = (*pProperties)[ELEMENT_LOOP_TOLERANCE];
+    if (std::abs(mTolerance) < std::numeric_limits<double>::epsilon()){
+        KRATOS_WARNING("FiberBeamColumnSection") <<
+            "WARNING: TOLERANCE FOR ELEMENT LOOP IS NOT SET." << std::endl;
+    }
+    KRATOS_CATCH("")
+}
+
+void FiberBeamColumnSection::Initialize()
+{
+    for (FiberBeamColumnUniaxialFiber& r_fiber : mFibers) {
+        r_fiber.Initialize();
+    }
+    UpdateLocalFlexibilityMatrix();
+}
+
+void FiberBeamColumnSection::GetGlobalFlexibilityMatrix(Matrix& rGlobalFlexibilityMatrix)
+{
+    KRATOS_TRY
+    rGlobalFlexibilityMatrix = ZeroMatrix(5, 5);
+    Matrix aux_matrix  = ZeroMatrix(5, 3);
+    Matrix b_matrix;
+    CalculateBMatrix(b_matrix);
+    (aux_matrix) += prod(Matrix(trans(b_matrix)), mLocalFlexibilityMatrix);
+    (rGlobalFlexibilityMatrix) += prod(aux_matrix, b_matrix);
+    KRATOS_CATCH("")
+}
+
+void FiberBeamColumnSection::UpdateLocalFlexibilityMatrix()
+{
+    KRATOS_TRY
+    // allocate memory
+    Matrix local_stiffness = ZeroMatrix(3, 3);
+    // get local 3x3 stiffness from fibers' tangential stiffness
+    for (FiberBeamColumnUniaxialFiber& r_fiber : mFibers) {
+        Matrix fiber_global_stiffness;
+        r_fiber.CreateGlobalFiberStiffnessMatrix(fiber_global_stiffness);
+        (local_stiffness) += fiber_global_stiffness;
+    }
+    // invert stiffness to get 3x3 flexibility
+    double det_stiffness = MathUtils<double>::Det(local_stiffness);
+    MathUtils<double>::InvertMatrix3(local_stiffness, mLocalFlexibilityMatrix, det_stiffness);
+    KRATOS_CATCH("")
+}
+
+void FiberBeamColumnSection::CalculateBMatrix(Matrix& rBMatrix)
+{
+    KRATOS_TRY
+    rBMatrix = ZeroMatrix(3, 5);
```

Comment:
it is possible to assign manually the zeros? (is cheaper), otherwise:
```suggestion
    noalias(rBMatrix) = ZeroMatrix(3, 5);
```

Reply:
This is just to ensure resizing the matrix and fill it to zeros. I pass a matrix of 0,0 size then resize it with zeros then fill the numbers. `noalias` would not work because it requires size conformity. Would it be a better solution to do `resize` then fill manually or simply pass the correct size of the matrix? This thing is also done in a couple other places in the code ..

Reply:
OK, in that case you should check the size first, resize if required and use noalias then

Reply:
Will do. Thanks